src/components/HoverPopup/HoverPopup.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+import {Popup} from '@gravity-ui/uikit';
+import debounce from 'lodash/debounce';
+
+import {cn} from '../../utils/cn';
+
+import './HoverPopup.scss';
+
+const b = cn('hover-popup');
+
+const DEBOUNCE_TIMEOUT = 100;
+
+interface HoverPopupProps {
+    children: React.ReactNode;
+    popupContent: React.ReactNode;
+    showPopup?: boolean;
+    offset?: [number, number];
+    anchorRef?: React.RefObject<HTMLElement>;
+    onShowPopup?: VoidFunction;
+    onHidePopup?: VoidFunction;
+}
+
+export const HoverPopup = ({
+    children,
+    popupContent,
+    showPopup,
+    offset,
+    anchorRef,
+    onShowPopup,
+    onHidePopup,
+}: HoverPopupProps) => {
+    const [isPopupVisible, setIsPopupVisible] = React.useState(false);
+    const anchor = React.useRef<HTMLDivElement>(null);
+
+    const debouncedHandleShowPopup = React.useMemo(
+        () =>
+            debounce(() => {
+                setIsPopupVisible(true);
+                onShowPopup?.();
+            }, DEBOUNCE_TIMEOUT),
+        [onShowPopup],
+    );
+
+    const hidePopup = React.useCallback(() => {
+        setIsPopupVisible(false);
+        onHidePopup?.();
+    }, [onHidePopup]);
+
+    const debouncedHandleHidePopup = React.useMemo(
+        () => debounce(hidePopup, DEBOUNCE_TIMEOUT),
+        [hidePopup],
+    );
+
+    const onMouseEnter = debouncedHandleShowPopup;
+
+    const onMouseLeave = () => {
+        debouncedHandleShowPopup.cancel();
+        debouncedHandleHidePopup();
+    };
+
+    const [isPopupContentHovered, setIsPopupContentHovered] = React.useState(false);
+    const [isFocused, setIsFocused] = React.useState(false);
+
+    const onPopupMouseEnter = React.useCallback(() => {
+        setIsPopupContentHovered(true);
+    }, []);
+
+    const onPopupMouseLeave = React.useCallback(() => {
+        setIsPopupContentHovered(false);
+    }, []);
+
+    const onPopupContextMenu = React.useCallback(() => {
+        setIsFocused(true);
+    }, []);
+
+    const onPopupBlur = React.useCallback(() => {
+        setIsFocused(false);
+    }, []);
+
+    const onPopupEscapeKeyDown = React.useCallback(() => {
+        setIsFocused(false);
+        setIsPopupContentHovered(false);
+        hidePopup();
+    }, [hidePopup]);
+
+    const open = isPopupVisible || showPopup || isPopupContentHovered || isFocused;
+
+    return (
+        <React.Fragment>
+            <div ref={anchor} onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
+                {children}
+            </div>
+            <Popup
+                contentClassName={b()}
+                anchorRef={anchorRef || anchor}
```

Comment:
Should we allow to pass `anchorRef`? Won't it break behaviour set by us?

Reply:
We allow to pass anchorRef for custom anchors if they are needed
if anchorRef is not passed - default anchor from component is used